src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionTests.cs

```diff
@@ -1362,6 +1377,447 @@ void M()
+                invokeKind: LSP.VSInternalCompletionInvokeKind.Typing,
+                triggerCharacter: "1",
+                triggerKind: LSP.CompletionTriggerKind.TriggerCharacter);
+
+            var results = await RunGetCompletionsAsync(testLspServer, completionParams).ConfigureAwait(false);
+            Assert.Contains(results.Items, item => item.Label == "dotnet_diagnostic.CS0021.severity");
+        }
+
+        [Fact]
+        public async Task TestCodeStyleSettingsNameInEditorconfigFiles()
+        {
+            var markup = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"" FilePath=""z:\TestProject\Assembly1.csproj"">
+        <Document FilePath=""File1.cs"">
+class C { }
+        </Document>
+        <AnalyzerConfigDocument FilePath=""z:\TestProject\.editorconfig"">
+unused_parameters{|caret:|}
+        </AnalyzerConfigDocument>
+    </Project>
+</Workspace>";
+            using var testLspServer = await CreateXmlTestLspServerAsync(markup, initializationOptions: new InitializationOptions { ServerKind = WellKnownLspServerKinds.EditorConfigLspServer, ClientCapabilities = s_completionCapabilities });
+            var p = testLspServer.GetCurrentSolution().Projects.Single().WithAnalyzerReferences(new[] { TestAnalyzerReferences });
+            await testLspServer.TestWorkspace.ChangeSolutionAsync(p.Solution).ConfigureAwait(false);
+
+            var caretLocation = testLspServer.GetLocations("caret").Single();
+            await testLspServer.OpenDocumentAsync(caretLocation.Uri);
+
+            var completionParams = CreateCompletionParams(
+                caretLocation,
+                invokeKind: LSP.VSInternalCompletionInvokeKind.Typing,
+                triggerCharacter: "s",
+                triggerKind: LSP.CompletionTriggerKind.TriggerCharacter);
+
+            var results = await RunGetCompletionsAsync(testLspServer, completionParams).ConfigureAwait(false);
+            Assert.Contains(results.Items, item => item.Label == "dotnet_code_quality_unused_parameters");
+        }
+
+        [Fact]
+        public async Task TestDontShowSettingsNameInEditorconfigFiles()
+        {
+            var markup = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"" FilePath=""z:\TestProject\Assembly1.csproj"">
+        <Document FilePath=""File1.cs"">
+class C { }
+        </Document>
+        <AnalyzerConfigDocument FilePath=""z:\TestProject\.editorconfig"">
+indent_size d{|caret:|}
+        </AnalyzerConfigDocument>
+    </Project>
+</Workspace>";
+            using var testLspServer = await CreateXmlTestLspServerAsync(markup, initializationOptions: new InitializationOptions { ServerKind = WellKnownLspServerKinds.EditorConfigLspServer, ClientCapabilities = s_completionCapabilities });
+            var p = testLspServer.GetCurrentSolution().Projects.Single().WithAnalyzerReferences(new[] { TestAnalyzerReferences });
```

Comment:
lets add a test helper method in this class that creates the server and updates the workspace solution with the test analyzer references so we don't need to duplciate this